### PR TITLE
Use named fields for enriching grpc info

### PR DESCRIPTION
**Pull request by mohit-a21**
Updated data models to have Rpc event field with system
value set to grpc in case the request is grpc. Added the check
rpc.system = grpc in SpanTypeAttributeEnricher.
Updated UserAgentSpanEnricher and ApiBoundryTypeAttributeEnricher
to use rpc.request.metadata. prefixed tags for identifying relevant
info. Also updated ApiStatusEnricher to use new named fields for
grpc status.

### CODE REVIEW

File: hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/ApiStatusEnricher.java
Comment (jcchavezs) [inline hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/ApiStatusEnricher.java:64]: I think once we are here it makes sense to define `Response response = event.getGrpc().getResponse()` (thanks Java) so we avoid chaining methods over and over and improve readability, other than that it is quite hard to read the method. Does it make sense?

File: hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/UserAgentSpanEnricher.java
Comment (jcchavezs) [inline hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/UserAgentSpanEnricher.java:73]: Same here, calling the attributes all the time makes it hard to read.

File: hypertrace-trace-enricher-impl/src/test/java/org/hypertrace/traceenricher/enrichment/enrichers/SpanTypeAttributeEnricherTest.java
Comment (jcchavezs) [inline hypertrace-trace-enricher-impl/src/test/java/org/hypertrace/traceenricher/enrichment/enrichers/SpanTypeAttributeEnricherTest.java:30]: I am curious about this test. In https://github.com/hypertrace/hypertrace-trace-enricher/pull/36/files#diff-9cc8b5577cab457f08f2a2e0ec514d26R181 we added quite some `if`s but the test class hasn't changed too much, maybe a table test would do the trick?

Comment (mohit-a21) [reply hypertrace-trace-enricher-impl/src/test/java/org/hypertrace/traceenricher/enrichment/enrichers/SpanTypeAttributeEnricherTest.java:30]: The change is a very small one. But for some weird reason the function location has changed. I'll move the function to old location and that should generate the correct diff

File: hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/ApiStatusEnricher.java
Comment (skjindal93) [reply hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/ApiStatusEnricher.java:32]: Can we also move such methods to the bottom of the class? Since, these are utility methods, we usually follow to keep them at the bottom of the class and have the actual logic at the top?

File: hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/SpanTypeAttributeEnricher.java
Comment (skjindal93) [inline hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/SpanTypeAttributeEnricher.java:79]: Just curious, why the method shift? Can we keep it here and have the utility methods at the bottom of the class, for better readability of the actual logic in the class?

File: hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/SpanTypeAttributeEnricher.java
Comment (skjindal93) [inline hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/SpanTypeAttributeEnricher.java:76]: Can you please check if this method is used anywhere else? If no, can we please change it to private? Similarly, for the method `getGrpcProtocol`

File: hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/SpanTypeAttributeEnricher.java
Comment (mohit-a21) [reply hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/SpanTypeAttributeEnricher.java:52]: Think nothing of it. I thought I found a powerful weapon in ⌥⌘L but sadly it has lead me to this.

File: hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/ApiStatusEnricher.java
Comment (mohit-a21) [reply hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/ApiStatusEnricher.java:43]: Done.

File: hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/ApiBoundaryTypeAttributeEnricher.java
Comment (mohit-a21) [reply hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/ApiBoundaryTypeAttributeEnricher.java:124]: I agree that the function name is not clear. But I don't think we should move enrichment out of the code path as that would lead to duplicated code.

Comment (mohit-a21) [reply hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/ApiBoundaryTypeAttributeEnricher.java:124]: Renamed method to `isHostHeaderEnrichmentSuccessful`

File: hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/SpanTypeAttributeEnricher.java
Comment (mohit-a21) [reply hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/SpanTypeAttributeEnricher.java]: This seems to be getting used in the test class.

File: hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/SpanTypeAttributeEnricher.java
Comment (mohit-a21) [reply hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/SpanTypeAttributeEnricher.java]: Bad formatting change. Moved the method to the original place.

File: hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/ApiStatusEnricher.java
Comment (mohit-a21) [reply hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/ApiStatusEnricher.java]: Done

File: hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/ApiBoundaryTypeAttributeEnricher.java
Comment (mohit-a21) [reply hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/ApiBoundaryTypeAttributeEnricher.java]: Same as above.

File: hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/ApiBoundaryTypeAttributeEnricher.java
Comment (mohit-a21) [reply hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/ApiBoundaryTypeAttributeEnricher.java]: Done.

File: hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/UserAgentSpanEnricher.java
Comment (mohit-a21) [reply hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/UserAgentSpanEnricher.java:74]: Done/

File: hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/ApiBoundaryTypeAttributeEnricher.java
Comment (skjindal93) [reply hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/ApiBoundaryTypeAttributeEnricher.java]: I don't see multiple invocations of `addEnrichedAttribute(event, HOST_HEADER, AttributeValueCreator.create(host));` as duplicated code. But, that could have been avoided as well by creating a new method `enrichHostHeader` which just calls the `addEnrichedAttribute` inside

Comment (skjindal93) [reply hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/ApiBoundaryTypeAttributeEnricher.java]: `isHostHeaderEnrichmentSuccessful` -> is also a misleading name. That means the enrichment on the span failed due to some reasons, instead of saying that enrich host header with valid host header

Comment (mohit-a21) [reply hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/ApiBoundaryTypeAttributeEnricher.java]: created a method `getSanitizedHostHeader`

**Review by jcchavezs — DISMISSED** [review]

**Comment by skjindal93** [comment]
Looks good to me. Waiting on dependent changes to be merged for the approval

File: hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/ApiBoundaryTypeAttributeEnricher.java
Comment (buchi-busireddy) [inline hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/ApiBoundaryTypeAttributeEnricher.java:150]: This can directly call `getSanitizedHostValue()` so that you return the final sanitized value.

File: hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/ApiStatusEnricher.java
Comment (buchi-busireddy) [inline hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/ApiStatusEnricher.java:84]: Extract this as a constant since it's not changing from span to span.

File: hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/SpanTypeAttributeEnricher.java
Comment (buchi-busireddy) [inline hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/SpanTypeAttributeEnricher.java:174]: Should this be the first check, above the block above?

File: hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/UserAgentSpanEnricher.java
Comment (buchi-busireddy) [inline hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/UserAgentSpanEnricher.java:74]: Should we check the protocol instead?

File: enriched-span-constants/build.gradle.kts
Comment (avinashkolluru) [inline enriched-span-constants/build.gradle.kts:63]: SNAPSHOT?

File: hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/ApiStatusEnricher.java
Comment (buchi-busireddy) [inline hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/ApiStatusEnricher.java:92]: I think this should be outside the `else` block

File: enriched-span-constants/build.gradle.kts
Comment (mohit-a21) [reply enriched-span-constants/build.gradle.kts:63]: This change is dependent on another change. Once that is merged, this will be updated.

File: hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/ApiBoundaryTypeAttributeEnricher.java
Comment (mohit-a21) [reply hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/ApiBoundaryTypeAttributeEnricher.java:150]: Done

File: hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/ApiStatusEnricher.java
Comment (mohit-a21) [reply hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/ApiStatusEnricher.java:84]: Done

File: hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/ApiStatusEnricher.java
Comment (mohit-a21) [reply hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/ApiStatusEnricher.java:92]: This is outside the else block.

File: hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/SpanTypeAttributeEnricher.java
Comment (mohit-a21) [reply hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/SpanTypeAttributeEnricher.java:174]: Moved this.

File: hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/UserAgentSpanEnricher.java
Comment (mohit-a21) [reply hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/UserAgentSpanEnricher.java:74]: Added protocol check.

**Review by tim-mwangi — APPROVED** [review]

**Pull request merged**
